### ISSUE TITLE
The n_refs argument ignored in Rust

### ISSUE THREAD

**Issue by dvukolov** [milesgranger/gap_statistic#45]
Hi Miles, it seems that the `n_refs` argument is silently ignored when using the Rust backend. This leads to additional discrepancies with the Python version.

**Comment by milesgranger** [comment]
oh that is sloppy.. left a TODO and everything
https://github.com/milesgranger/gap_statistic/blob/08519d84f87fbee9ac32cdc0851b84c9b36c6395/src/gap_statistic/mod.rs#L54

Thanks for the reminder!

**Comment by shahabh** [comment]
It seems that n_refs argument is still ignored. I passed 100 but it didn't increase the runtime and the result is very random (optimalK changes every time I run it).

**Comment by milesgranger** [comment]
Hi, thanks for raising the issue.

After trying around a bit, it seems that `n_refs` is being passed; confirmed by following the code and that runtime for me certainly does change as I increase `n_refs`

I do concede that optimalk changes much more than when using 'joblib' and thus the scipy kmeans as the clustering algorithm. Which in a way makes some sense; there are some tunable settings that aren't tunable in the current implementation, and it would not surprise me if I messed up somewhere along the lines in my implementation.

So in short, I guess it's good that the `rust` backend is now optional, as one _probably_ won't want to use that, or at least seems quite sensitive to the data distribution itself. I would advise you to perhaps use a different backend/clusterer if possible.

Sorry for the trouble. :+1:
